CRM/Member/BAO/Query.php

```diff
@@ -209,7 +209,8 @@ public static function whereClauseSingle(&$values, &$query) {
         $strtolower = function_exists('mb_strtolower') ? 'mb_strtolower' : 'strtolower';
         $value = $strtolower(CRM_Core_DAO::escapeString(trim($value)));
 
-        $query->_where[$grouping][] = "civicrm_membership.source $op '{$value}'";
```

Comment:
@jitendrapurohit does it work with this function ? 

https://github.com/civicrm/civicrm-core/blob/a43deb747f3d0fab276f923140c85c24aef76284/CRM/Contact/BAO/Query.php#L7233

Ideally that function should be able to replace most chunks like this....

Also - can you remove the strtolower stuff above as well - we got rid of most of it but I guess that bit is still hanging in there.

Reply:
By removing, do you mean to remove the "if mb.."? or to remove lowercasing completely? (I'm not familiar with the topic, just curious, but the "if" could definitely be removed, as long as we default to `mb_strtolower`.

Reply:
remove lowercasing completely

Reply:
We used to lower case in php & then use the index-bypassing LOWER to make it really slow but now we rely on php case-insensitivity

Reply:
I've updated the PR to use `handleWhereFromMetadata()` for setting the where clause. Thanks 👍